pulp_2_tests/tests/docker/api_v2/test_remove.py

```diff
@@ -50,19 +75,51 @@ def setUp(self):
                 'enable_v1': False,
                 'enable_v2': True,
                 'feed': DOCKER_V2_FEED_URL,
-                'upstream_name': get_upstream_name(self.cfg),
+                'upstream_name': UPSTREAM_NAME,
             }
         )
         self.repo = self.client.post(REPOSITORY_PATH, body)
         self.addCleanup(self.client.delete, self.repo['_href'])
         sync_repo(self.cfg, self.repo)
 
+    def create_and_copy_test_repo(self, source_repo, copy_units):
+        """Return repo of repos for test."""
+        body = gen_repo(
+            importer_config={
+                'enable_v1': False,
+                'enable_v2': True,
+            }
+        )
+        repo = self.client.post(REPOSITORY_PATH, body)
+        self.copy_docker_units(repo, source_repo, copy_units)
+        return repo
+
+    def copy_docker_units(self, repo, source_repo, units):
+        """Copy specified docker units."""
+        criteria = {
+            'type_ids': [
+                'docker_tag',
+                'docker_manifest_list',
+                'docker_manifest',
+                'docker_blob',
+            ],
+            'filters': {'unit': {'digest': {"$in": units}}},
+        }
+        self.client.post(
+            urljoin(repo['_href'], 'actions/associate/'),
+            {'source_repo_id': source_repo['id'], 'criteria': criteria},
+        )
+
     def get_docker_units(self, repo, unit_type):
```

Comment:
It seems that we want the number of units per type, if that is the right thinking perhaps...
```suggestion
    def get_docker_units_count(self, repo, unit_type):
```

Reply:
Agreed. In the middle of action.